File: src/07.greedy/01.arrayGreedy/16.646.go
package array_greedy

import "sort"

func FindLongestChain(pairs [][]int) int {
	n := len(pairs)

	sort.Slice(pairs, func(i, j int) bool {
		return pairs[i][1] < pairs[j][1]
	})

	res := 1
	pos := pairs[0][1]

	for i := 1; i < n; i++ {
		if pos < pairs[i][0] {
			res++
			pos = pairs[i][1]
		}
	}

	return res
}

File: src/07.greedy/01.arrayGreedy/16.646.ts
export function findLongestChain(pairs: number[][]): number {
    const n = pairs.length;

    pairs.sort((a, b) => a[1] - b[1]);

    let pos = pairs[0][1];

    let res = 1;

    for (let i = 1; i < n; i++) {
        if (pos < pairs[i][0]) {
            res++;
            pos = pairs[i][1];
        }
    }

    return res;
}
